velox/connectors/hive/tests/HivePartitionUtilTest.cpp

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/connectors/hive/HivePartitionUtil.h"
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/vector/tests/utils/VectorTestBase.h"
+
+#include "gtest/gtest.h"
+
+using namespace facebook::velox::connector::hive;
+using namespace facebook::velox;
+
+class HivePartitionUtilTest : public ::testing::Test,
+                              public test::VectorTestBase {
+ protected:
+  template <typename T>
+  VectorPtr makeDictionary(const std::vector<T>& data) {
+    auto base = makeFlatVector(data);
+    auto indices = makeIndices(data.size(), [](auto row) { return row; });
+    return wrapInDictionary(indices, data.size(), base);
+  };
+};
+
+TEST_F(HivePartitionUtilTest, partitionString) {
+  {
+    RowVectorPtr input = makeRowVector(
+        {"flat_bool_col",
+         "flat_tinyint_col",
+         "flat_smallint_col",
+         "flat_int_col",
+         "flat_bigint_col",
+         "dict_string_col",
+         "const_date_col"},
+        {makeFlatVector<bool>(std::vector<bool>{false}),
+         makeFlatVector<int8_t>(std::vector<int8_t>{10}),
+         makeFlatVector<int16_t>(std::vector<int16_t>{100}),
+         makeFlatVector<int32_t>(std::vector<int32_t>{1000}),
+         makeFlatVector<int64_t>(std::vector<int64_t>{10000}),
+         makeDictionary<StringView>(std::vector<StringView>{"str1000"}),
+         makeConstant<Date>(Date(10000), 1)});
+    std::vector<column_index_t> partitionChannels{0, 1, 2, 3, 4, 5, 6};
+
+    EXPECT_EQ(
+        makePartitionName(input, partitionChannels, 0),
```

Comment:
Perhaps, add a test case with a single partition key. That's the most common case.

Reply:
Revised the test to include the case of single partition.